File: frontend/src/objects/order/apiTypes/CheckoutRequest.ts
import type { VoucherId } from '@/objects/shared/ids'
import type { MerchantId } from '@/objects/shared/ids'

import type { CheckoutLine } from '../CheckoutLine'

export interface OrderMerchantNote {
  merchantId: MerchantId
  text?: string | null
  imageUrl?: string | null
}

export interface CheckoutRequest {
  lines: CheckoutLine[]
  customerName?: string
  customerPhone?: string
  deliveryAddress?: string
  voucherId?: VoucherId
  merchantNotes?: OrderMerchantNote[]
}

File: backend/src/order/objects/apiTypes/CheckoutRequest.scala
package delivery.order.objects.apiTypes

import delivery.order.objects.CheckoutLine
import delivery.shared.objects.VoucherId

final case class CheckoutRequest(
    lines: List[CheckoutLine],
    customerName: Option[String] = None,
    customerPhone: Option[String] = None,
    deliveryAddress: Option[String] = None,
    voucherId: Option[VoucherId] = None
)
